### ISSUE TITLE
Alfresco Full-Text Search Brings Back Incorrect Results when Phrasing

### ISSUE THREAD

**Issue by ZachGerth** [tsgrp/hpi#950]
When searching using the full-text search on Solr, using phrases (that is, surrounding a search term with quotation marks) will return every document; if part of a search is in quotes and part isn't, it will only filter on the parts that aren't in quotes. For example:

Searching for Document yields all documents with that term in it.
Searching for "Document" yields all documents, no exceptions.
Searching for "Document" blue car yields all documents with the terms 'blue' and 'car'.

Some ideas thrown around were to make it so if double quotes are used, don't append wildcards.

**Comment by gsteimer** [comment]
@benallenallen - will this be an issue for HBase?  What about Alfresco?  Or does this only apply to Solr portal impls?

**Comment by benallenallen** [comment]
The current impl doesn't seem to be suffering from this issue, this may just be from an old Solr client project... I just ran the following on edge3 for a document with a title of: 

`Disaster Recovery Document For Recovering from Disasters`

This is the query that was run:

`"Recovery Document"`

and it brought back the document correctly

**Comment by gsteimer** [comment]
Cool - @ZachGerth can you check the trunk codebase and see if there's evidence for this working properly in trunk?  If we can find something, then I think we can close this as invalid.

**Comment by ZachGerth** [comment]
Yeah - I just reviewed the trunk codebase and it should work fine on trunk, judging by the way it affects the full text search clauses.

**Comment by gsteimer** [comment]
I think there's definitely some issues with this on Alfresco.  Here's what I'm seeing.

**Content snippet - edge2, Doc: TSG-105**
![image](https://cloud.githubusercontent.com/assets/1377228/12356696/8db9a53e-bb69-11e5-8d70-388124d147cd.png)

**FT Search - search for word daily - correctly returned**
![image](https://cloud.githubusercontent.com/assets/1377228/12356724/b2f0b1e4-bb69-11e5-95cd-450c803777d3.png)

**FT Search - search for phrase daily operations (no quotes) - correctly returned**
![image](https://cloud.githubusercontent.com/assets/1377228/12356755/d536d008-bb69-11e5-9e1e-6b731fc2d3ae.png)

**FT Search - search for phrase daily operations (with quotes) - correctly returned**
![image](https://cloud.githubusercontent.com/assets/1377228/12356782/fe85e7aa-bb69-11e5-9f7c-28ffb0ab6bfb.png)

Here's where it get's interesting...

**FT Search - search for phrase daily checklist (no quotes) - no results returned (incorrect)**
Here, I'd expect TSG-105 to come back based on what happens in Dctm, Solr, and HBase.  The two words are in the document, just not right next to each other
![image](https://cloud.githubusercontent.com/assets/1377228/12356813/2a7fdbcc-bb6a-11e5-9c31-f2d033795c79.png)

Similar results when searching for _Daily maintenance_:
![image](https://cloud.githubusercontent.com/assets/1377228/12356887/927e9d6c-bb6a-11e5-869b-57522b4486b1.png)


**FT Search - search for phrase daily checklist (with quotes) - no results returned (correctly)**
Obviously, we shouldn't get results for this one (the two words are not right next to each other anywhere in the document), but if we fix the one above, we shuold also make sure this search doesn't bring back the doc.
![image](https://cloud.githubusercontent.com/assets/1377228/12356828/4673c1ea-bb6a-11e5-8ed4-1de6a8bb42bd.png)

**Comment by gsteimer** [comment]
For reference, this is the query OC is running (from the OC logs):

```
Generated Query:  TYPE:aw\:qualityDocument AND ALL:"Daily Checklist\*" AND -ASPECT:cm\:workingcopy
```

Edge2 is Alfresco 4.2.3.1, so I'm assuming the ALL keyword
For reference, this is the query OC is running (from the OC logs):

```
Generated Query:  TYPE:aw\:qualityDocument AND ALL:"Daily Checklist\*" AND -ASPECT:cm\:workingcopy
```

**Comment by benallenallen** [comment]
That is the problem:

```ALL:"Daily Checklist"```

is looking for the exact phrase "Daily Checklist". I can see why it was written that way because that was the only way to get the query to actually run 'correctly' for phrases that had spaces in it. My understanding of the `"ALL:"` keyword is that it searches against every individual field AND the 'text' field. If that is the case, we _could_ take out the quotes, split on spaces and individually "AND" each separate word (this is how old school Solr portals used to work):

`ALL: "Daily" AND ALL: "Checklist"`

The problem we have with this is that we would have to add a bunch of logic if we wanted to be able to support users actually WANTING to do an exact search with quotes and make it act differently.

@parzgnat would probably be able to help understand better with a 'quick' conversation, but how the code is written right now clearly won't work.

**Comment by gsteimer** [comment]
@benallenallen @parzgnat - if I remember right, the `ALL` keyword went away in Alfresco 5.  If the new query syntax fixes this issue in Alfresco 5, then maybe we can change this to an 'Upon Client Request' item.

**Comment by benallenallen** [comment]
I think it is going to be the same issue on Alfresco 5 the way we are building our query:

```TEXT:"Daily Checklist" OR d:text:"Daily Checklist" OR d:mltext:"Daily Checklist"```

is what we generate in Alfresco 5. As soon as we put quotes around it that is where we get into trouble.

**Comment by gsteimer** [comment]
Hmm.. so maybe the fix for both is to remove the hardcoded quotes around what the user puts in?

**Comment by benallenallen** [comment]
I believe we would just change the quotes to parentheses... can't get rid of them entirely as that causes different behaviour.

**Comment by grahamsinger** [comment]
@peterkk Can this be closed with OC revs 13722, 13723, 13732 and 13734?

**Comment by peterkk** [comment]
Closed with hpi revision #13734
